awx/ui_next/src/screens/ManagementJob/ManagementJobList/ManagementJobListItem.jsx

```diff
@@ -0,0 +1,126 @@
+import React, { useState } from 'react';
+import { withI18n } from '@lingui/react';
+import { t } from '@lingui/macro';
+import { Link, useHistory } from 'react-router-dom';
+import { Button, Tooltip } from '@patternfly/react-core';
+import { Tr, Td } from '@patternfly/react-table';
+import { RocketIcon } from '@patternfly/react-icons';
+
+import { SystemJobTemplatesAPI } from '../../../api';
+import AlertModal from '../../../components/AlertModal';
+import ErrorDetail from '../../../components/ErrorDetail';
+import { ActionsTd, ActionItem } from '../../../components/PaginatedTable';
+import LaunchManagementPrompt from './LaunchManagementPrompt';
+
+function ManagementJobListItem({
+  i18n,
+  onLaunchError,
+  isPrompted,
+  isSuperUser,
+  id,
+  name,
+  description,
+}) {
+  const detailsUrl = `/management_jobs/${id}`;
+
+  const history = useHistory();
+  const [isLaunchLoading, setIsLaunchLoading] = useState(false);
+
+  const [isManagementPromptOpen, setIsManagementPromptOpen] = useState(false);
+  const [isManagementPromptLoading, setIsManagementPromptLoading] = useState(
+    false
+  );
+  const [managementPromptError, setManagementPromptError] = useState(null);
+  const handleManagementPromptClick = () => setIsManagementPromptOpen(true);
+  const handleManagementPromptClose = () => setIsManagementPromptOpen(false);
+
+  const handleManagementPromptConfirm = async days => {
+    setIsManagementPromptLoading(true);
+    try {
+      const { data } = await SystemJobTemplatesAPI.launch(id, {
+        extra_vars: { days },
+      });
+      history.push(`/jobs/management/${data.id}/output`);
+    } catch (error) {
+      setManagementPromptError(error);
+    } finally {
+      setIsManagementPromptLoading(false);
+    }
+  };
+
+  const handleLaunch = async () => {
+    setIsLaunchLoading(true);
+    try {
+      const { data } = await SystemJobTemplatesAPI.launch(id);
+      history.push(`/jobs/management/${data.id}/output`);
+    } catch (error) {
+      onLaunchError(error);
+    } finally {
+      setIsLaunchLoading(false);
+    }
+  };
+
+  return (
+    <>
+      <Tr id={`mgmt-jobs-row-${id}`}>
```

Comment:
If it's possible, can we have the job name as a selector here? Just in case of changing job ID numbers.

Reply:
@one-t I've added identifiers that use the `job_type` instead of the id:

```
cleanup_activitystream
cleanup_tokens
cleanup_sessions
cleanup_jobs
```

You'll see the id's rendered as: `mgmt-jobs-row-cleanup-sessions`, etc.:

![Screenshot from 2021-02-23 11-19-54](https://user-images.githubusercontent.com/9753817/108874548-423bf000-75ca-11eb-8fc9-c35ea2907cb6.png)